### PR TITLE
fix the bug compressed type of bool and fix array

### LINKED ISSUE #392: failed to create compressed slog file on Fixed Array types.
#include <string>
#include <hobbes/hobbes.H>
#include <hobbes/storage.H>
#include <hobbes/db/file.H>
#include <hobbes/db/series.H>
#include <hobbes/lang/tylift.H>

typedef unsigned char fixarray[16];

DEFINE_STRUCT(
  FixarrayS,
  (fixarray, id)
);

int main()
{
  hobbes::cc c; 

  hobbes::writer writer{"/var/tmp/test.log"};

  std::cout << "=============Create  Raw Stream [OK]" << std::endl;
  hobbes::series<FixarrayS> ss1(&c, &writer, "udata1", 10000, hobbes::StoredSeries::Raw);
  FixarrayS v1;
  ss1(v1);

  std::cout << "=============Create Compressed Stream [Failed]" << std::endl;
  hobbes::series<FixarrayS> ss2(&c, &writer, "udata2", 10000, hobbes::StoredSeries::Compressed);
  FixarrayS v2;
  ss2(v2);

  return 0;
}

### PR DESCRIPTION
**Pull request by dawa79**
fix #392

### CODE REVIEW

File: test/Storage.C [hunk 1]
```diff
@@ -164,6 +174,8 @@ TEST(Storage, CSeriesAPI) {
       st.x = i;
       st.y = 3.14159 * static_cast<double>(i);
       st.z = makeString("string_" + str::from(i));
+      st.b = true;
```
Comment (smunix) [inline test/Storage.C:177]: let's test bools and fixed-size arrays?
`
st.b = (i % 2) == 0 
`

Comment (smunix) [reply test/Storage.C:177]: `
EXPECT_TRUE(c.compileFn<bool()>("[b|{b=b}<-cf.series_test] == take(1000,cycle([true, false]))")());
`
You would need to define `cycle` if it doesn't exist yet

**Review by smunix — CHANGES_REQUESTED** [review]
This is looking good, thank you Dan! Let's have a few more tests in it

File: test/Storage.C (hunk 1 above)
Comment (dawa79) [reply test/Storage.C]: I have update the test case.  

st.z = makeString("string_" + str::from(i));
 st.b = i % 2 == 0;
 memcpy(st.v, st.z->data, 8);

 EXPECT_TRUE(c.compileFn<bool()>("[b|{b=b}<-cf.series_test][:0] == [x%2==0|{x=x}<-cf.series_test][:0]")());
 EXPECT_TRUE(c.compileFn<bool()>("[v|{v=v}<-cf.series_test][:0] == [z[0:8]|{z=z}<-cf.series_test][:0]")());

Comment (smunix) [reply test/Storage.C]: nice, thanks!

**Review by smunix — APPROVED** [review]

**Pull request merged**